src/engraving/dom/engravingitem.cpp

```diff
@@ -1314,6 +1314,33 @@ void EngravingItem::checkVoiceAssignmentCompatibleWithTrack()
     }
 }
 
+bool EngravingItem::elementAppliesToTrack(const track_idx_t refTrack) const
+{
+    if (!hasVoiceAssignmentProperties()) {
+        return refTrack == track();
+    }
+
+    const VoiceAssignment voiceAssignment = getProperty(Pid::VOICE_ASSIGNMENT).value<VoiceAssignment>();
+    if (voiceAssignment == VoiceAssignment::CURRENT_VOICE_ONLY && track() == refTrack) {
+        return true;
+    }
+
+    if (voiceAssignment == VoiceAssignment::ALL_VOICE_IN_STAFF && track2staff(track()) == track2staff(refTrack)) {
+        return true;
+    }
+
+    const Part* elementPart = part();
+    if (!elementPart) {
+        return false;
+    }
+
+    if (voiceAssignment == VoiceAssignment::ALL_VOICE_IN_INSTRUMENT && (elementPart->startTrack() <= refTrack
+                                                                        && elementPart->endTrack() - 1 >= refTrack)) {
+        return true;
+    }
+    return false;
+}
```

Comment:
This code is repeated identical in Spanner::elementAppliesToTrack in order to check for both track() and track2(). Here's a suggestion for how to improve it.

Move this bit of code into a protected static function, something like:
```
protected:
static bool elementAppliesToTrack(track_idx_t elementTrack, track_idx_t refTrack, VoiceAssignment voiceAssignment);
```

Then you can have:
```
bool EngravingItem::appliesToTrack(track_idx_t refTrack) const
{
    if (!hasVoiceAssignmentProperties()) {
        return refTrack == track();
    }
    return elementAppliesToTrack(track(), refTrack, voiceAssignment());
}
```
and the Spanner version as:
```
bool Spanner::appliesToTrack(track_idx_t refTrack) const
{
    if (!hasVoiceAssignmentProperties()) {
        return refTrack == track() || refTrack == track2();
    }
    return elementAppliesToTrack(track(), refTrack, voiceAssignment()) || elementAppliesToTrack(track2(), refTrack, voiceAssignment());
}
```